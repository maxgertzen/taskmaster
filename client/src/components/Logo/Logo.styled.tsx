import styled from '@emotion/styled';

import { Title } from '../Title/Title';

export const LogoContainer = styled.div<{ withTitle?: boolean }>(
  ({ theme, withTitle = false }) => ({
    display: 'flex',
    alignItems: 'center',
    ...(withTitle ? {} : { justifyContent: 'center' }),
    svg: {
      width: theme.spacing(4),
      height: theme.spacing(4),
      marginRight: theme.spacing(1),
    },
    path: {
      fill: theme.colors.primary,
    },
  })
);

export const StyledTitle = styled(Title)<{
  size: 'small' | 'medium' | 'large';
}>(({ theme, size }) => ({
  fontSize:
    size === 'small'
      ? theme.typography.h6.fontSize
      : size === 'medium'
        ? theme.typography.h4.fontSize
        : theme.typography.h1.fontSize,
}));
