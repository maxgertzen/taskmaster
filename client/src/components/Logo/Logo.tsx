import { FC } from 'react';

import taskMasterGrayScaleUrl from '../../assets/taskmaster-grayscale.png';
import taskMasterUrl from '../../assets/taskmaster.png';
import { useThemeStore } from '../../store/themeStore';

import {
  LogoContainer,
  StyledTitle,
  TaskMasterLogoComponent,
} from './Logo.styled';

interface LogoProps {
  size?: 'small' | 'medium' | 'large';
  withTitle?: boolean;
}

export const Logo: FC<LogoProps> = ({ size = 'medium', withTitle = false }) => {
  const themeMode = useThemeStore((state) => state.theme);
  return (
    <LogoContainer>
      <TaskMasterLogoComponent
        src={themeMode === 'dark' ? taskMasterGrayScaleUrl : taskMasterUrl}
        alt='TaskMaster Logo'
      />
      {withTitle && (
        <StyledTitle size={size} variant='h1'>
          TaskMaster
        </StyledTitle>
      )}
    </LogoContainer>
  );
};
