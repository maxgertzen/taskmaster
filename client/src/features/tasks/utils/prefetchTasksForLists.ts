import { QueryClient } from '@tanstack/react-query';

import { QUERY_KEYS } from '@/shared/api/query-keys';
import { STALE_TIME } from '@/shared/constants/staleTime';

import { fetchTasks } from '../api';

export const prefetchTasksForLists = (
  listIds: string[],
  queryClient: QueryClient,
  token: string | null
) => {
  if (!token || !listIds.length) return;

  listIds.forEach((listId) => {
    queryClient.prefetchQuery({
      queryKey: QUERY_KEYS.tasks({ listId }),
      queryFn: fetchTasks(token),
      staleTime: STALE_TIME,
    });
  });
};
