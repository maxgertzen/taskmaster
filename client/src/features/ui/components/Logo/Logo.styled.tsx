import styled from '@emotion/styled';

export const LogoContainer = styled('div')<{ withTitle?: boolean }>(
  ({ theme, withTitle = false }) => ({
    display: 'flex',
    alignItems: 'center',
    ...(withTitle ? {} : { justifyContent: 'center' }),
    svg: {
      width: theme.spacing(4),
      height: theme.spacing(4),
      marginRight: theme.spacing(1),
    },
    path: {
      fill: theme.colors.primary.main,
    },
  })
);

export const StyledTitle = styled('h1')<{
  size: 'small' | 'medium' | 'large';
}>(({ theme, size }) => ({
  fontSize:
    size === 'small'
      ? theme.typography.h6.fontSize
      : size === 'medium'
        ? theme.typography.h4.fontSize
        : theme.typography.h1.fontSize,
  [`@media (max-width: ${theme.breakpoints.sm}px)`]: {
    display: 'none',
  },
}));

export const TaskMasterLogoComponent = styled.img(({ theme }) => ({
  width: theme.spacing(12),
  height: theme.spacing(10),
}));
