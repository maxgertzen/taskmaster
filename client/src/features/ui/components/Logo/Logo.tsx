import { FC } from 'react';

import taskMasterGrayScaleUrl from '@/shared/assets/images/taskmaster-grayscale.png';
import taskMasterUrl from '@/shared/assets/images/taskmaster.png';
import { useThemeStore } from '@/shared/store/themeStore';

import {
  LogoContainer,
  StyledTitle,
  TaskMasterLogoComponent,
} from './Logo.styled';

interface LogoProps {
  size?: 'small' | 'medium' | 'large';
  withTitle?: boolean;
}

export const Logo: FC<LogoProps> = ({ size = 'medium', withTitle = false }) => {
  const themeMode = useThemeStore((state) => state.theme);
  return (
    <LogoContainer>
      <TaskMasterLogoComponent
        data-testid='logo'
        src={themeMode === 'dark' ? taskMasterGrayScaleUrl : taskMasterUrl}
        alt='TaskMaster Logo'
      />
      {withTitle && <StyledTitle size={size}>TaskMaster</StyledTitle>}
    </LogoContainer>
  );
};
