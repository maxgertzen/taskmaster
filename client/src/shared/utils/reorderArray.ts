export const reorderArray = <T>(
  array: T[],
  fromIndex: number,
  toIndex: number
): T[] => {
  const reordered = [...array];
  const [movedItem] = reordered.splice(fromIndex, 1);
  reordered.splice(toIndex, 0, movedItem);
  return reordered;
};
