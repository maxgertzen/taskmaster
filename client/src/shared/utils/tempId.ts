export const generateTempId = (prefix: string = 'temp') =>
  `${prefix}-${Date.now()}`;
