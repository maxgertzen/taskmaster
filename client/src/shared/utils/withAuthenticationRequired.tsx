import { withAuthenticationRequired as realWithAuthenticationRequired } from '@auth0/auth0-react';

import { IS_AUTH0_DISABLED } from '@/constants';
import { mockWithAuthenticationRequired } from '@/mocks';

const isMock = IS_AUTH0_DISABLED === 'true';

export const withAuthenticationRequired = isMock
  ? mockWithAuthenticationRequired
  : realWithAuthenticationRequired;
