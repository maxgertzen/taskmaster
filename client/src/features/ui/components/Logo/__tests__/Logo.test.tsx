import { describe, it, expect, vi, beforeEach } from 'vitest';

import {
  mockThemeStore,
  mockThemeStoreMode,
} from '@/tests/__mocks__/storeMocks';
import { render, screen } from '@/tests/utils';

import { Logo } from '../Logo';

vi.mock('@/shared/store/themeStore', () => ({
  useThemeStore: mockThemeStore,
}));

vi.mock('@/shared/assets/images/taskmaster.png', () => ({
  default: 'taskmaster.png',
}));

vi.mock('@/shared/assets/images/taskmaster-grayscale.png', () => ({
  default: 'taskmaster-grayscale.png',
}));

describe('Logo', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  describe.each([
    ['light', 'taskmaster.png'],
    ['dark', 'taskmaster-grayscale.png'],
  ])('in %s theme', (theme, expectedSrc) => {
    beforeEach(() => {
      mockThemeStoreMode.mockReturnValue(theme);
    });

    it.each([
      ['small', false],
      ['medium', false],
      ['large', false],
      ['small', true],
      ['medium', true],
      ['large', true],
    ])('renders correctly with size=%s and withTitle=%s', (size, withTitle) => {
      const { container } = render(
        <Logo
          size={size as 'small' | 'medium' | 'large'}
          withTitle={withTitle}
        />
      );
      expect(container).toMatchSnapshot();
    });

    it('uses correct logo image for theme', () => {
      render(<Logo />);
      const logo = screen.getByTestId('logo');
      expect(logo).toHaveAttribute('src', expectedSrc);
    });
  });

  describe('Title Rendering', () => {
    beforeEach(() => {
      mockThemeStoreMode.mockReturnValue('light');
    });

    it('shows title when withTitle is true', () => {
      render(<Logo withTitle={true} />);
      expect(screen.getByText('TaskMaster')).toBeInTheDocument();
    });

    it('hides title when withTitle is false', () => {
      render(<Logo withTitle={false} />);
      expect(screen.queryByText('TaskMaster')).not.toBeInTheDocument();
    });
  });
});
