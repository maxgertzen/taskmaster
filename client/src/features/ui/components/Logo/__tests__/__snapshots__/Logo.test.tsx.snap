// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Logo > in dark theme > renders correctly with size=large and withTitle=false 1`] = `
<div>
  <div
    class="css-v7ss51"
  >
    <img
      alt="TaskMaster Logo"
      class="css-fhjm1v"
      data-testid="logo"
      src="taskmaster-grayscale.png"
    />
  </div>
</div>
`;

exports[`Logo > in dark theme > renders correctly with size=large and withTitle=true 1`] = `
<div>
  <div
    class="css-v7ss51"
  >
    <img
      alt="TaskMaster Logo"
      class="css-fhjm1v"
      data-testid="logo"
      src="taskmaster-grayscale.png"
    />
    <h1
      class="css-1obiie8"
    >
      TaskMaster
    </h1>
  </div>
</div>
`;

exports[`Logo > in dark theme > renders correctly with size=medium and withTitle=false 1`] = `
<div>
  <div
    class="css-v7ss51"
  >
    <img
      alt="TaskMaster Logo"
      class="css-fhjm1v"
      data-testid="logo"
      src="taskmaster-grayscale.png"
    />
  </div>
</div>
`;

exports[`Logo > in dark theme > renders correctly with size=medium and withTitle=true 1`] = `
<div>
  <div
    class="css-v7ss51"
  >
    <img
      alt="TaskMaster Logo"
      class="css-fhjm1v"
      data-testid="logo"
      src="taskmaster-grayscale.png"
    />
    <h1
      class="css-dytuun"
    >
      TaskMaster
    </h1>
  </div>
</div>
`;

exports[`Logo > in dark theme > renders correctly with size=small and withTitle=false 1`] = `
<div>
  <div
    class="css-v7ss51"
  >
    <img
      alt="TaskMaster Logo"
      class="css-fhjm1v"
      data-testid="logo"
      src="taskmaster-grayscale.png"
    />
  </div>
</div>
`;

exports[`Logo > in dark theme > renders correctly with size=small and withTitle=true 1`] = `
<div>
  <div
    class="css-v7ss51"
  >
    <img
      alt="TaskMaster Logo"
      class="css-fhjm1v"
      data-testid="logo"
      src="taskmaster-grayscale.png"
    />
    <h1
      class="css-nczlas"
    >
      TaskMaster
    </h1>
  </div>
</div>
`;

exports[`Logo > in light theme > renders correctly with size=large and withTitle=false 1`] = `
<div>
  <div
    class="css-v7ss51"
  >
    <img
      alt="TaskMaster Logo"
      class="css-fhjm1v"
      data-testid="logo"
      src="taskmaster.png"
    />
  </div>
</div>
`;

exports[`Logo > in light theme > renders correctly with size=large and withTitle=true 1`] = `
<div>
  <div
    class="css-v7ss51"
  >
    <img
      alt="TaskMaster Logo"
      class="css-fhjm1v"
      data-testid="logo"
      src="taskmaster.png"
    />
    <h1
      class="css-1obiie8"
    >
      TaskMaster
    </h1>
  </div>
</div>
`;

exports[`Logo > in light theme > renders correctly with size=medium and withTitle=false 1`] = `
<div>
  <div
    class="css-v7ss51"
  >
    <img
      alt="TaskMaster Logo"
      class="css-fhjm1v"
      data-testid="logo"
      src="taskmaster.png"
    />
  </div>
</div>
`;

exports[`Logo > in light theme > renders correctly with size=medium and withTitle=true 1`] = `
<div>
  <div
    class="css-v7ss51"
  >
    <img
      alt="TaskMaster Logo"
      class="css-fhjm1v"
      data-testid="logo"
      src="taskmaster.png"
    />
    <h1
      class="css-dytuun"
    >
      TaskMaster
    </h1>
  </div>
</div>
`;

exports[`Logo > in light theme > renders correctly with size=small and withTitle=false 1`] = `
<div>
  <div
    class="css-v7ss51"
  >
    <img
      alt="TaskMaster Logo"
      class="css-fhjm1v"
      data-testid="logo"
      src="taskmaster.png"
    />
  </div>
</div>
`;

exports[`Logo > in light theme > renders correctly with size=small and withTitle=true 1`] = `
<div>
  <div
    class="css-v7ss51"
  >
    <img
      alt="TaskMaster Logo"
      class="css-fhjm1v"
      data-testid="logo"
      src="taskmaster.png"
    />
    <h1
      class="css-nczlas"
    >
      TaskMaster
    </h1>
  </div>
</div>
`;
